### ISSUE TITLE
Can't trigger CalDAV update on sync (endless loading)

### ISSUE THREAD

**Issue by TekkertheChaot** [FossifyOrg/Calendar#270]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Calendar/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Calendar/issues) or [closed](https://github.com/FossifyOrg/Calendar/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Calendar/discussions) or [closed](https://github.com/FossifyOrg/Calendar/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Calendar/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.3

### Affected Android/Custom ROM version

Stock Samsung Android 14 / OneUI 6.1

### Affected device model

Samsung Galaxy Flip 3

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

_Setup:
Using Nextcloud as a CalDAV server and DAVx⁵ for managing DAV resources (connector to nextcloud in this case)_

1. in nextcloud: create an event in a synced calendar.
2. in the app: go to the day the event was created on the server
3. trigger a sync / update of the calendar
4. --> Forever loading...

### Expected behavior

Finish loading, update the calendar

### Actual behavior

loads forever, never updates the calendar

### Screenshots/Screen recordings
_(sorry for low bitrate, had to compress it to fit the upload size imit)_
https://github.com/FossifyOrg/Calendar/assets/17234581/38079b73-4771-4a70-93e9-97c3973c68ff

### Additional information

other calendar apps like SPlanner are able to trigger a CalDAV update / sync
Here is a logcat: 
[logcat_11072024_1352.log](https://github.com/user-attachments/files/16176886/logcat_11072024_1352.log)

**Comment by Aga-C** [comment]
Related to https://github.com/FossifyOrg/Calendar/issues/193.

**Comment by TekkertheChaot** [comment]
Ahhh okay i overlooked that. I read the title and dismissed it prematurely... Should i close this one as a duplicate?

**Comment by Aga-C** [comment]
It's not a duplicate. I just linked these two issues because they are similar to each other and may have the same root cause. If it were a duplicate, I'd close it 🙂

**Comment by mpxx1** [comment]
Dev: Samsung Galaxy A32
Using Nextcloud as a CalDAV server and DAVx⁵ for managing DAV resources (connector to nextcloud in this case)
Installed from F-Droid, version 1.0.3
Android 11

Had the same issue, but it's gone after i switched off and on 'CalDAV sync' button in calendar settings
After that davx5 sync have also start working for calendar, but if i call sync inside app (pulling down reload) it shows massage that it's successfully done, but it's actually not.

**Comment by opa-rudi** [comment]
I also have this problem on a rooted crdroid Samsung Galaxy S20. Manual syncing with davx5 when swiping down in the app results in stucking (endless loading). It looks like the app is unable to trigger a davx5 sync. I have to wait until davx5 itself updates the data.

**Comment by goldnetcoding** [comment]
Related to #656.

**Comment by goldnetcoding** [comment]
I believe the root cause is permission-related.

Currently, the app constructs `Account` instances manually:
https://github.com/FossifyOrg/Calendar/blob/ea40162f57221b15e2ced8b48822d1f5966483ab/app/src/main/kotlin/org/fossify/calendar/extensions/Context.kt#L923-L941

It retrieves `accountName` and `accountType` from calendar entries using the `READ_CALENDAR` permission, without resolving the accounts through `AccountManager`:
https://github.com/FossifyOrg/Calendar/blob/ea40162f57221b15e2ced8b48822d1f5966483ab/app/src/main/kotlin/org/fossify/calendar/helpers/CalDAVHelper.kt#L86-L122

It appears that the app is calling `ContentResolver.requestSync()` with `Account` objects for accounts that are not visible to the app without the Contacts permission. This results in the sync not being triggered and the UI remaining in an endless loading state, similar to #656.

### Reproduction
Tested on:
- Pixel 9 running GrapheneOS (Android 16)
- Samsung Galaxy A54 running One UI 8.0

Steps to reproduce:
1. Ensure the app does **not** have the Contacts permission granted (e.g., revoke it in system settings).
2. Open the app and manually trigger a calendar refresh.
3. The refresh enters an endless loading state and sync is not performed.

### Verification
1. Grant the Contacts permission in system settings.
2. Trigger a manual refresh again.
3. Sync completes successfully.

### Suggested Fix
Request the `GET_ACCOUNTS` permission (which is part of the `CONTACTS` permission group) when enabling CalDAV sync and when manually refreshing. If the permission is denied, display a clear error message to the user.

**Comment by goldnetcoding** [comment]
> Request the `GET_ACCOUNTS` permission (which is part of the `CONTACTS` permission group)

After further testing, requesting the `GET_ACCOUNTS` permission (which I initially believed would be more relevant to account visibility) does not resolve the issue on all devices. The sync call succeeds only after the `READ_CONTACTS` permission is granted on my Samsung device and in an Android emulator. The `GET_ACCOUNTS` permission was not necessary.

On my Pixel 9 running GrapheneOS, the `GET_ACCOUNTS` permission alone was sufficient for the sync call to succeed. This indicates that the behavior varies across device configurations and system builds.

I confirmed this by temporarily adding the `READ_SYNC_SETTINGS` permission and using `ContentResolver.getIsSyncable()` to inspect sync availability for the calendar authority.

**Comment by goldnetcoding** [comment]
After some research, I found that [starting with Android 8 (API 26)](https://developer.android.com/about/versions/oreo/android-8.0-changes#aaad), the intended and supported way to gain access to user accounts is through `AccountManager.newChooseAccountIntent()`, which explicitly requests user consent.

I have tested and confirmed that this approach makes the accounts syncable without requesting any Contacts-related permissions. My previous suggestion should be considered a workaround. Using the `AccountManager` is the correct and future-proof solution.

**Comment by naveensingh** [comment]
> I have tested and confirmed that this approach makes the accounts syncable without requesting any Contacts-related permissions. My previous suggestion should be considered a workaround. Using the AccountManager is the correct and future-proof solution.

You are correct, but I'm curious if you have tried making account authenticators visible to the app:

```xml
        <intent>
            <action android:name="android.accounts.AccountAuthenticator" />
        </intent>
```

This workaround/patch was used in https://github.com/FossifyOrg/Contacts/commit/5cfa250db873619888e2d5dd268d5196e3438e73 to fix https://github.com/FossifyOrg/Contacts/issues/83.

**Comment by goldnetcoding** [comment]
> You are correct, but I'm curious if you have tried making account authenticators visible to the app

I just tried that in an emulator, but it did not resolve this issue. This issue is related to account access restrictions, not package visibility restrictions. 
While your referenced workaround addresses a broader scope for #656, my PR #1091 is more specific, targeting only the sync adapter visibility required to fix that particular sync problem.

My suggestion would still be to allow managing granted accounts explicitly through a list in the UI, where the user can grant or revoke access per account at any time, using the `AccountManager`. Instead of using a modal dialog, this could be implemented as a nested account detail screen with a proper list of calendars for that account, allowing them to be managed individually rather than combining everything into one large list.
It could also make the UI clearer by showing the account provider icon for each account, which would help users distinguish between multiple accounts that use the same email address but different providers.

I understand that this would likely be more of a feature request, but it is still relevant to properly resolving this issue. The only workaround that consistently worked across all of my tested devices was the one I mentioned earlier (requesting the `READ_CONTACTS` permission). As noted previously, the permission behavior appeared to differ across devices (`GET_ACCOUNTS` being sufficient on a device running GrapheneOS, while `READ_CONTACTS` appeared to fix it on all tested devices), so it is unclear whether this workaround will remain reliable in the long term.
